### PR TITLE
Add allow_auto_merge support to Repository

### LINKED ISSUE #2476: Add support of `allow_auto_merge` attribute to Repository
Adds the `allow_auto_merge` attribute to a Repository: https://docs.github.com/en/rest/repos/repos#update-a-repository

> `allow_auto_merge` boolean
> Either `true` to allow auto-merge on pull requests, or `false` to disallow auto-merge.

This attribute can be modified via edit (on a private repository):

```
>>> g.get_repo("<test repo>").allow_auto_merge
True
>>> g.get_repo("<test repo>").edit(allow_auto_merge=False)
>>> g.get_repo("<test repo>").allow_auto_merge
False
>>> g.get_repo("<test repo>").edit(allow_auto_merge=True)
>>> g.get_repo("<test repo>").allow_auto_merge
True
```

### PR DESCRIPTION
**Pull request by BradChengIRESS**
Closes #2476 

Adds `allow_auto_merge` attribute to a Repository: https://docs.github.com/en/rest/repos/repos#update-a-repository

> `allow_auto_merge` boolean
> Either `true` to allow auto-merge on pull requests, or `false` to disallow auto-merge.

This attribute can be modified via edit (on a private repository):

```
>>> g.get_repo("<test repo>").allow_auto_merge
True
>>> g.get_repo("<test repo>").edit(allow_auto_merge=False)
>>> g.get_repo("<test repo>").allow_auto_merge
False
>>> g.get_repo("<test repo>").edit(allow_auto_merge=True)
>>> g.get_repo("<test repo>").allow_auto_merge
True
```

### CODE REVIEW

**Review by JLLeitschuh — APPROVED** [review]
LGTM!

**Comment by testworksau** [comment]
@s-t-e-v-e-n-k are the maintainers the only reviewers with the ability to approve PRs?

https://github.com/PyGithub/PyGithub/blob/master/MAINTAINERS

**Comment by JLLeitschuh** [comment]
I believe so

**Comment by EnricoMi** [comment]
I can also approve and merge PRs. Right now I am waiting for the hotfix v1.58.1 to be released before I merge new features into master. That release is now taking more than a month, hilarious: https://github.com/PyGithub/PyGithub/discussions/2453#discussioncomment-5663908

I am happy to have a look here soon.

**Comment by EnricoMi** [comment]
It looks like it, maintainers are not very communicative.

File: tests/ReplayData/Repository.testEditWithAllArguments.txt
Comment (testworksau) [inline tests/ReplayData/Repository.testEditWithAllArguments.txt:7]:
```suggestion
{"has_wiki": false, "name": "PyGithub", "has_downloads": true, "has_issues": true, "homepage": "http://vincent-jacques.net/PyGithub", "private": true, "description": "Description edited by PyGithub", "has_projects": false, "allow_auto_merge": true, "allow_update_branch": true, "allow_forking": true, "allow_squash_merge": true, "allow_merge_commit": true, "allow_rebase_merge": true, "delete_branch_on_merge": true}
```

**Review by JLLeitschuh — APPROVED** [review]
Still LGTM! @EnricoMi feel free to merge

**Pull request merged**
